include/lstk/lstk.hpp

```diff
@@ -139,6 +139,15 @@ static inline std::vector<std::string> split_on_get_strings(std::string_view s,
     return ret;
 }
 
+static inline std::vector<std::string_view> split_on_first(std::string_view s, char delim)
+{
+    auto first_delim = s.find(delim);
+    if (first_delim == std::string_view::npos)
+        throw std::logic_error("Attempt to split string on nonexistent delimiter");
```

Comment:
Why not just return the original string here? if we want to handle both cases we'll have to catch an exception here